### DISCUSSION TITLE
Custom function for selecting all rows.

### DISCUSSION THREAD

**Discussion by husnainali92** [TanStack/table#4603] [Q&A]
Hi I'm using react aria [checkbox](https://react-spectrum.adobe.com/react-aria/useCheckbox.html) for selection. 
Their onChange method doesn't send the native event when onChange triggers they only send isSelected (e.target.checked) flag.
But getToggleAllRowsSelectedHandler and getToggleAllPageRowsSelectedHandler expect event. 

My app crashed when I try to select all and react table was reading checked property. Screenshot attached.

![image](https://user-images.githubusercontent.com/101558118/207292396-d63c2c9f-7868-4631-9a5b-048b39f4d22b.png)

The weird thing is single row selection is working fine but select all not working. They both have different function but both functions are expecting event. 

Here is the codesandbox [link ](https://codesandbox.io/s/react-table-selection-si6hzi?file=/src/main.tsx) to see

**Comment by rickvian** [comment]
Hi, i bumped to this issue as well, due to my component passing boolean of checked instead of event

it looks like the ideal fix is to make the component passes the event

but here is the work around

because it try to access e.target.checked, you can create such structure, passes it to function that returned by getToggleAllRowsSelectedHandler

```
  onChange: (checked) =>
    table?.getToggleAllRowsSelectedHandler()?.({
      target: { checked },
    }),
```

**Reply by nmatsumoto4** [reply]
I also solved it using this method. It seems that getToggleAllRowsSelectedHandler assumes a DOM event format. When using a custom component other than a primitive checkbox, it needs to be handled as mentioned above.

**Comment by KevinVandy** [comment]
Just don't use the get handlers APIs. They are just shortcuts for the toggle function APIs.

**Reply by codenoobforreal** [reply]
> Just don't use the get handlers APIs. They are just shortcuts for the toggle function APIs.

Is it `toggleAllRowsSelected` method the right choice.
It works well with RAC.
![image](https://github.com/user-attachments/assets/95d7fa39-e019-4528-8a14-460ac6529ba1)
